src/RpcClient.ts

```diff
@@ -646,81 +650,118 @@ export class RpcClient {
+        minContextSlot,
+      };
+    }
+    legacyTransaction = new Transaction().add(...instructions);
+    legacyTransaction.recentBlockhash = recentBlockhash;
+    legacyTransaction.feePayer = payerKey;
 
-      if (feePayer) {
-        legacyTransaction.partialSign(feePayer);
-      }
+    for (const signer of signers) {
+      legacyTransaction.partialSign(signer);
+    }
 
-      return { smartTransaction: legacyTransaction, lastValidBlockHeight };
+    if (feePayer) {
+      legacyTransaction.partialSign(feePayer);
     }
+
+    return {
+      txBuff: legacyTransaction.serialize(),
+      blockhash,
+      minContextSlot,
+    };
   }
 
   /**
    * Build and send an optimized transaction, and handle its confirmation status
    * @param {TransactionInstruction[]} instructions - The transaction instructions
    * @param {Signer[]} signers - The transaction's signers. The first signer should be the fee payer
    * @param {AddressLookupTableAccount[]} lookupTables - The lookup tables to be included in a versioned transaction. Defaults to `[]`
-   * @param {SendOptions & { feePayer?: Signer }} sendOptions - Options for sending the transaction, including an optional feePayer. Defaults to `{ skipPreflight: false }`
+   * @param {SendOptions & { feePayer?: Signer; lastValidBlockHeightOffset?: number }} sendOptions - Options for sending the transaction, including an optional feePayer and lastValidBlockHeightOffset. Defaults to `{ skipPreflight: false; lastValidBlockheightOffset: 150 }`
    * @returns {Promise<TransactionSignature>} - The transaction signature
    */
   async sendSmartTransaction(
     instructions: TransactionInstruction[],
     signers: Signer[],
     lookupTables: AddressLookupTableAccount[] = [],
-    sendOptions: SendOptions & { feePayer?: Signer } = { skipPreflight: false }
+    sendOptions: SendOptions & {
+      feePayer?: Signer;
+      lastValidBlockHeightOffset: number;
+    } = { skipPreflight: false, lastValidBlockHeightOffset: 150 }
   ): Promise<TransactionSignature> {
+    if (sendOptions.lastValidBlockHeightOffset < 0)
+      throw new Error('expiryBlockOffset must be a positive integer');
+
     try {
       // Create a smart transaction
-      const { smartTransaction: transaction, lastValidBlockHeight } =
+      const { txBuff, blockhash, minContextSlot } =
         await this.createSmartTransaction(
           instructions,
           signers,
           lookupTables,
           sendOptions.feePayer
         );
 
-      // Timeout of 60s. The transaction will be routed through our staked connections and should be confirmed by then
-      const timeout = 60000;
-      const startTime = Date.now();
-      let txtSig;
-
-      while (
-        Date.now() - startTime < timeout ||
-        (await this.connection.getBlockHeight()) <= lastValidBlockHeight
-      ) {
+      const recdSendOptions: SendOptions = {
+        maxRetries: 0,
+        preflightCommitment: 'confirmed',
+        skipPreflight: true,
```

Comment:
We should be consistent here with the default values for `skipPreflight` — in the function's params we have it set to `false` however it's set to `true` here

Reply:
Note that you do not need to skip preflight checks to be routed through staked connections. So, I'm more inclined that we respect the user's `skipPreflight` settings if provided, otherwise default to `false`